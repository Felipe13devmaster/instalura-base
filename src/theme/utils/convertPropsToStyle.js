import { breakpointsMedia } from "./breakpointsMedia";

export function convertPropsToStyle(propName) {
    return (props) => {//Sempre que retornamos uma função, o styled-component executa essa função
        const propValue = props[propName];//prop value pode ser string ou objeto
        if (typeof propValue === 'string') {
            return {
                //textAlign: props.textAlign === linha 20
                [propName]: props[propName]//[]: indica que este cara é a chave do objeto
            }
        }

        if (typeof propValue === 'object') {
            return breakpointsMedia({
                xs: {
                    [propName]: propValue.xs
                },
                sm: {
                    [propName]: propValue.sm
                },
                md: {
                    [propName]: propValue.md
                },
                lg: {
                    [propName]: propValue.lg
                },
                xl: {
                    [propName]: propValue.xl
                },
            });
        }
    }
}