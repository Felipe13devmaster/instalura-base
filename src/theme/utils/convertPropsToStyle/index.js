import breakpointsMedia from '../breakpointsMedia';

function convertPropsToStyle(propName) {
  // eslint-disable-next-line consistent-return
  return (props) => { // Sempre que retornamos uma função, o styled-component executa essa função
    const propValue = props[propName];// prop value pode ser string ou objeto
    if (typeof propValue === 'string' || typeof propValue === 'number') {
      return {
        // textAlign: props.textAlign === linha 20
        [propName]: props[propName], // []: indica que este cara é a chave do objeto
      };
    }

    if (typeof propValue === 'object') {
      const breakpoints = {};

      if (propValue.xs) breakpoints.xs = { [propName]: propValue.xs };
      if (propValue.sm) breakpoints.sm = { [propName]: propValue.sm };
      if (propValue.md) breakpoints.md = { [propName]: propValue.md };
      if (propValue.lg) breakpoints.lg = { [propName]: propValue.lg };
      if (propValue.xl) breakpoints.xl = { [propName]: propValue.xl };

      return breakpointsMedia(breakpoints);
    }
    return {};
  };
}

export default convertPropsToStyle;
