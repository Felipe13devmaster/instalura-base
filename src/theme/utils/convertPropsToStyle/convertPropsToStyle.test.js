import convertPropsToStyle from './index';

describe('convertPropsToStyle()', () => {
  describe('Quando receber um argumanto simples', () => {
    test('é uma string', () => {
      const convertPropsToStyleResultado = convertPropsToStyle('textAlign');
      const componentProps = { textAlign: 'center' };
      const styleResultado = convertPropsToStyleResultado(componentProps);
      // console.log(convertPropsToStyleResultado(componentProps));
      expect(styleResultado).toEqual({ textAlign: 'center' });
    });
    test('é um numero', () => {
      const convertPropsToStyleResultado = convertPropsToStyle('flex');
      const componentProps = { flex: 1 };
      const styleResultado = convertPropsToStyleResultado(componentProps);
      expect(styleResultado).toEqual({ flex: 1 });
    });
  });

  describe('quando receber apenas um argumento com breakingpoints', () => {
    test('renders only one breakpoint resolution', () => {
      const propToStyleResult = convertPropsToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center' } };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
    test('renders two or more breakpoint resolutions', () => {
      const propToStyleResult = convertPropsToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center', md: 'right' } };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
  });
});
