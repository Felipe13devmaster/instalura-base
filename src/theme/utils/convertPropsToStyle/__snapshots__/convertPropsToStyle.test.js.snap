// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertPropsToStyle() quando receber apenas um argumento com breakingpoints renders only one breakpoint resolution 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`convertPropsToStyle() quando receber apenas um argumento com breakingpoints renders two or more breakpoint resolutions 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "text-align: right;",
    "}",
  ],
]
`;
